include/swift/Basic/ExponentialGrowthAppendingBinaryByteStream.h

```diff
@@ -77,7 +77,7 @@ class ExponentialGrowthAppendingBinaryByteStream
       Data.resize(RequiredSize);
     }
 
-    *(T *)(Data.data() + Offset) = Value;
+    ::memcpy(Data.data() + Offset, &Value, sizeof Value);
```

Comment:
If this is undefined behaviour, we should just revert https://github.com/apple/swift/pull/18955. I will create a revert PR. Thanks for catching this!

Reply:
It is, but I’m pretty sure memcpy() in a template function like this is different from memcpy() in an implementation file. If you rerun whatever benchmarks convinced you that writeRaw() was a good idea with this implementation of it, I suspect you’ll see similar speed improvements to the unaligned one.

Reply:
Thanks for the hint. I will take a look.

Reply:
I stopped my revert. Merging this change would be great.